src/analysis/processing/qgsalgorithmsavefeatures.cpp

```diff
@@ -0,0 +1,119 @@
+/***************************************************************************
+                         qgsalgorithmsavefeatures.cpp
+                         ---------------------
+    begin                : July 2020
+    copyright            : (C) 2020 by Mathieu Pellerin
+    email                : nirvn dot asia at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmsavefeatures.h"
+
+///@cond PRIVATE
+
+QString QgsSaveFeaturesAlgorithm::name() const
+{
+  return QStringLiteral( "savefeatures" );
+}
+
+QString QgsSaveFeaturesAlgorithm::displayName() const
+{
+  return QObject::tr( "Save vector features to file" );
+}
+
+QStringList QgsSaveFeaturesAlgorithm::tags() const
+{
+  return QObject::tr( "save,write" ).split( ',' );
+}
+
+QString QgsSaveFeaturesAlgorithm::group() const
+{
+  return QObject::tr( "Vector general" );
+}
+
+QString QgsSaveFeaturesAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectorgeneral" );
+}
+
+QString QgsSaveFeaturesAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "Save vector features to a specified file dataset." );
```

Comment:
```suggestion
  return QObject::tr( "This algorithm saves vector features to a specified file dataset." );
```

+ some description about the dataset/layer options, and where to find help for them